pkg/client/client.go

```diff
@@ -12,3 +19,34 @@ type VMClient interface {
 	StartVM(id string) error
 	Close() error
 }
+
```

Comment:
This should stay in the provider code. In this package we should only have interfaces.

Reply:
The `newReconciler` function was doing `factory := ovirtprovider.NewSourceClientFactory()` directly. It made sense to me to move `SourceClientFactory` and `NewSourceClientFactory` into a shared place so that `newReconciler` could do `client.NewSourceClientFactory()`, which would support both providers. How would you recommend I do this instead?

Reply:
Thinking a bit more about it we can move it here as you suggested.